### DISCUSSION TITLE
Metrics HELP conflict results in data loss

### DISCUSSION THREAD

**Discussion by ZhouZhongzhi** [prometheus/client_golang#985] [Q&A]
My client has two versions of node_exporter in their system environment
The HELP for node_cpu_seconds_total is different in the two versions

```
# HELP node_cpu_seconds_total Seconds the CPUs spent in each mode.
# HELP node_cpu_seconds_total Seconds the cpus spent in each mode.
```

If both node_exporter indicators are collected at the same time
One of these will be discarded due to HELP conflicts

Why do we do this?
It is only the difference of HELP

**Comment by ZhouZhongzhi** [comment]
ErrorInfo：

`collected metric node_cpu_seconds_total label:<name:"cpu" value:"9" > label:<name:"instance" value:"192.168.101.1:9100" > label:<name:"instance_ip" value:"192.168.101.1" > label:<name:"job" value:"test" > label:<name:"mode" value:"system" >  counter:<value:230265.5 > timestamp_ms:1644227543046  has help "Seconds the CPUs spent in each mode." but should have "Seconds the cpus spent in each mode."`

**Comment by ZhouZhongzhi** [comment]
@beorn7

**Comment by beorn7** [comment]
The Prometheus exposition format doesn't allow different HELP strings for one and the same metric name. Therefore, an instrumentation library has to resolve the conflict _somehow_. It is safest to bail out with a clear error. And it would be quite weird to ignore "small" errors like typos but then have some threshold (if the string is "totally different") where we would throw the error again.

Within the same binary, you have full control of the exposed metrics and should be able to fix the problem.

I don't quite understand the premise here, though, i.e. "My client has two versions of node_exporter in their system environment". Every node_exporter should run on their own. They should not end up in the same client_golang registry, unless you somehow take the code of both node_exporter versions and use both code bases in one and the same derived binary.
